js/modules/k6/http/http_request_test.go

```diff
@@ -626,6 +635,20 @@ func TestRequestAndBatch(t *testing.T) {
 				assert.NoError(t, err)
 				assertRequestMetricsEmitted(t, state.Samples, "GET", sr("HTTPBIN_IP_URL/digest-auth/auth/bob/pass"), url, 200, "")
 			})
+			t.Run("ntlm", func(t *testing.T) {
+				ntlmServer := httptest.NewServer(http.HandlerFunc(ntlmHandler("bob", "pass")))
+				defer ntlmServer.Close()
+
+				state.Samples = nil
+				url := strings.Replace(ntlmServer.URL, "http://", "http://bob:pass@", -1)
+
+				_, err := common.RunString(rt, fmt.Sprintf(`
+				let res = http.request("GET", "%s", null, { auth: "ntlm" });
```

Comment:
I agree it would be good more tests, but to do that, requires me to implement the NTLM mock as the spec says. Right now it is a very naive implementation, just enough to give a basic validation, implement it properly it is not a trivial task and I'm not sure we should spend even more time in a feature that will be barely used.

Reply:
But I'll try to add as much tests as possible without having to implement more stuff in ntlm mock

Reply:
Yeah, very good points, it's not worth it for this :+1: